### DISCUSSION TITLE
Any chance of seeing a more compute like version of "CrestFillMaskArtefacts"?

### DISCUSSION THREAD

**Discussion by dkyeck** [crest-ocean/crest#1070] [Q&A]
Please describe the desired feature/behaviour.
**CrestFillMaskArtefacts** right now uses a lot of load texture instructions.

Supporting images/mockups, or example reference videos are very helpful to communicate ideas.
using a more compute like approach utilizing _group shared memory_ should reduce the number of load texture instructions significantly.

**Comment by magenta404** [comment]
Tried it and performance was worse when measured using Nsight. But maybe I was doing something wrong. You can see it for yourself here: [perf/fill-mask-shared-memory](https://github.com/wave-harmonic/crest/tree/perf/fill-mask-shared-memory).
